WooCommerce/Classes/ViewRelated/Orders/Cells/SummaryTableViewCell.swift

```diff
@@ -51,21 +56,11 @@ final class SummaryTableViewCell: UITableViewCell {
 
     /// Displays the specified OrderStatus, and applies the right Label Style
     ///
-    func display(viewModel: OrderDetailsViewModel) {
-        if let orderStatus = viewModel.orderStatus {
-            paymentStatusLabel.applyStyle(for: orderStatus.status)
-            paymentStatusLabel.text = orderStatus.name
-        } else {
-            // There are unsupported extensions with even more statuses available.
-            // So let's use the order.statusKey to display those as slugs.
-            let statusKey = viewModel.order.statusKey
-            let statusEnum = OrderStatusEnum(rawValue: statusKey)
-            paymentStatusLabel.applyStyle(for: statusEnum)
-            paymentStatusLabel.text = viewModel.order.statusKey
-        }
+    func display(presentation: SummaryTableViewCellPresentation) {
+        paymentStatusLabel.applyStyle(for: presentation.status)
```

Comment:
This method removes the `else` statement that makes sure the unsupported custom status labels get styled correctly. Please restore the original if/else code.

How this cell should render unsupported, custom status labels:
<img width="354" alt="Screen Shot 2019-07-08 at 4 28 57 PM" src="https://user-images.githubusercontent.com/1062444/60843917-8ed87a80-a19d-11e9-96b8-74bf3fdfae66.png">

How this cell is currently rendering unsupported, custom status labels:
<img width="360" alt="Screen Shot 2019-07-08 at 4 29 03 PM" src="https://user-images.githubusercontent.com/1062444/60843945-a6176800-a19d-11e9-9929-86e56125d407.png">

Reply:
Nice catch, sorry about that. I tried (and failed) to remove that responsibility from the cell (notice how we used to pass the view model and now we pass something of type `SummaryTableViewCellPresentation`. If you don't mind, I'd prefer to give it another try and to fix my bug, and keep that logic out of the cell. If I can't I'll roll the change back.